File: src/main/java/no/nav/domain/pensjon/regler/vedtak/FortsattMedlemskap.java
package no.nav.domain.pensjon.regler.vedtak;

import no.nav.domain.pensjon.regler.grunnlag.Eksportforbud;
import no.nav.domain.pensjon.regler.grunnlag.Unntak;

/**
 * @author Magnus Bakken (Accenture), PK-10597, PKPYTON-1063
 * @author Steinar Hjellvik Decisive,  PK-6559
 */
public class FortsattMedlemskap extends AbstraktVilkar {


    /**
     * Fortsatt medlem i folketrygden
     */
    private Boolean fortsattMedlemFT;

    /**
     * Brukeren har minst 20 års botid i Norge
     */
    private Boolean minstTyveArBotidNorge;

    /**
     * Opptjent rett til tilleggspensjon etter folketrygdloven
     */
    private Boolean opptjentRettTilTPEtterFT;

    /**
     * Eksportforbud
     */
    private Eksportforbud eksportforbud;

    /**
     * Fri eksport fordi uførhet skyldes yrkesskade
     */
    private Boolean friEksportPgaYrkesskade;

    /**
     * Innvilget garantert tilleggspensjon til ung ufør/død skal eksporteres
     */
    private Boolean innvilgetGarantertTP;

    /**
     * Eksportrett garantert TP
     */
    private Unntak eksportrettGarantertTP;

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/vedtak/FortsattMedlemskap.kt
package no.nav.pensjon.regler.domain.vedtak

import no.nav.pensjon.regler.domain.grunnlag.Eksportforbud
import no.nav.pensjon.regler.domain.grunnlag.Unntak

class FortsattMedlemskap : AbstraktVilkar() {
    /**
     * Fortsatt medlem i folketrygden
     */
    var fortsattMedlemFT: Boolean? = null

    /**
     * Brukeren har minst 20 års botid i Norge
     */
    var minstTyveArBotidNorge: Boolean? = null

    /**
     * Opptjent rett til tilleggspensjon etter folketrygdloven
     */
    var opptjentRettTilTPEtterFT: Boolean? = null

    /**
     * Eksportforbud
     */
    var eksportforbud: Eksportforbud? = null

    /**
     * Fri eksport fordi uførhet skyldes yrkesskade
     */
    var friEksportPgaYrkesskade: Boolean? = null

    /**
     * Innvilget garantert tilleggspensjon til ung ufør/død skal eksporteres
     */
    var innvilgetGarantertTP: Boolean? = null

    /**
     * Eksportrett garantert TP
     */
    var eksportrettGarantertTP: Unntak? = null

}